log.go

```diff
@@ -0,0 +1,57 @@
+package marker
+
+import (
+	"io"
+	"os"
+
+	"github.com/fatih/color"
+)
+
+// StdoutMarkerOption is functional option type for StdoutMarker
+type StdoutMarkerOption func(*StdoutMarker)
+
+// MarkRule contains marking information to be applied on log stream
+type MarkRule struct {
+	Matcher MatcherFunc
+	Color   *color.Color
+}
+
+// StdoutMarker contains specified rules for applying them on output
+type StdoutMarker struct {
+	rules []MarkRule
+	out   io.Writer
+}
+
+// NewWriteMarker creates a Marker that writes out to the given io.Writer
+func NewWriteMarker(writer io.Writer, options ...StdoutMarkerOption) *StdoutMarker {
```

Comment:
hey @cyucelen in writing the docs, I refactored the way that we deal with custom `out`s. instead of having to replace stdout on the created marker, i flipped it so both cases (stdout and custom) require a single function call and no more than that.

let me know what you think, happy to discuss changing it back but it felt more natural when making the example file for this writer type so i thought it might be a good change